src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
Use reduce for Summation: Replaced forEach with reduce for more concise and functional summation logic.

`const sum = children.reduce((acc, ch) => {
  const mother = people.find((m) => m.name === ch.mother);
  return acc + (ch.born - mother.born);
}, 0);`